build.gradle

```diff
@@ -0,0 +1,221 @@
+plugins {
+    id 'java'
+    id 'java-library'
+    id 'maven-publish'
+    id 'signing'
+    id 'com.google.protobuf' version '0.8.15'
+    id 'io.github.gradle-nexus.publish-plugin' version '1.0.0'
+}
+
+group = 'com.exactpro.th2'
+version = release_version
+
+sourceCompatibility = 11
+targetCompatibility = 11
+
+ext {
+    grpcVersion = '1.32.1'
+    reactiveGrpcVersion = '1.0.1'
+    protobufVersion = '3.13.0'
+    serviceGeneratorVersion = '3.1.12'
+
+    genDir = file("${project.rootDir}/src/gen")
+    sharedDir = file("${project.rootDir}/shared")
+}
+
+repositories {
+    mavenCentral()
+
+    maven {
+        url 'https://s01.oss.sonatype.org/content/repositories/snapshots/'
+    }
+
+    maven {
+        url 'https://s01.oss.sonatype.org/content/repositories/releases/'
+    }
+
+    mavenLocal()
+
+    configurations.all {
+        resolutionStrategy.cacheChangingModulesFor 0, 'seconds'
+        resolutionStrategy.cacheDynamicVersionsFor 0, 'seconds'
+    }
+}
+
+dependencies {
+    api platform('com.exactpro.th2:bom:3.0.0')
+
+    api 'com.exactpro.th2:grpc-common:3.1.2'
+
+    implementation 'io.grpc:grpc-protobuf'
+    implementation 'io.grpc:grpc-core'
+    implementation 'io.grpc:grpc-netty'
+
+    api "com.salesforce.servicelibs:rxgrpc-stub:${reactiveGrpcVersion}"
+    implementation 'io.reactivex.rxjava2:rxjava:2.2.19'
+
+    implementation 'javax.annotation:javax.annotation-api:1.3.2'
+
+    api "com.exactpro.th2:grpc-service-generator:${serviceGeneratorVersion}"
+}
+
+sourceSets {
+    main {
+        proto {
+            srcDirs 'src/main/proto'
+        }
+
+        java {
+            srcDirs 'src/gen/main/java', 'src/gen/main/grpc', 'src/gen/main/rxgrpc', 'src/gen/main/services/java/src'
+        }
+
+        resources {
+            srcDirs 'src/gen/main/services/java/resources'
+        }
+    }
+}
+
+protobuf {
+    protoc {
+        artifact = "com.google.protobuf:protoc:${protobufVersion}"
+    }
+
+    plugins {
+        grpc {
+            artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}"
+        }
+
+        rxgrpc {
+            artifact = "com.salesforce.servicelibs:rxgrpc:${reactiveGrpcVersion}"
+        }
+
+        services {
+            artifact = "com.exactpro.th2:grpc-service-generator:${serviceGeneratorVersion}:all@jar"
+        }
+    }
+
+    generateProtoTasks.generatedFilesBaseDir = genDir
+
+    generateProtoTasks {
+        all()*.plugins {
+            grpc {}
+            rxgrpc {}
```

Comment:
Maybe redundant

Reply:
How can I check if it's redundant?

Reply:
We use rxjava3 in th2-conn instead of v2

Reply:
```We use rxjava3 in th2-conn instead of v2```

I don't get it. Could you please elaborate on that?

Reply:
The rxgrpc plugin (implementation 'io.reactivex.rxjava2:rxjava:2.2.19') generate stubs for using in a rxjava style. jar with the current grpc classes will have dependencies on rxjava2. 
Neither of our components uses rxjava2 and the additional dependency will be redundant in our case